pallets/contracts/src/lib.rs

```diff
@@ -281,6 +350,9 @@ decl_storage! {
             map hasher(identity) ExtrinsicId => bool;
         /// Storage version.
         StorageVersion get(fn storage_version) build(|_| Version::new(1)): Version;
+        /// Stores the upgrades suported for each contract given the api and chain version.
+        pub SupportedApiUpgrades get (fn api_tracker):
+            double_map hasher(identity) T::AccountId, hasher(twox_64_concat) Api => BoundedBTreeMap<ChainVersion, ApiCodeHash<T>, T::MaxApiUpgrades>;
```

Comment:
We don't want this to include `AccountId` of the contract.  That would require a transaction for each deployed contract.  We only want to maintain one set of code hashes for all contracts that use the same `Api` version.